auto_round/autoround.py

```diff
@@ -2616,6 +2641,29 @@ def quantize_blocks(
         clear_memory()
         input_ids = to_device(input_ids, self.cache_device)
         input_others = to_device(input_others, self.cache_device)
+        if self.recipe_mode:
```

Comment:
It would be better to wrap this new code into a function and call it as early as possible.